File: unity/Assets/Scripts/Debuff/Debuff_2.cs
using UnityEngine;
using System.Collections;
using System.Collections.Generic;

public class Debuff_2 : Debuff {
	override public void effect(){

		int damage = this.caster.mp;
		damage = this.caster.calcDamage(damage, this.character, 2);
		this.character.dot(new popupText(){type=1, value=damage}, this.caster);

		Dictionary<string, string> data = new Dictionary<string, string>();
		data.Add("prop", "2");
		data.Add("targetId", this.character.side.ToString() + '-' + this.character.charId.ToString());
		data.Add("targetTitle", this.character.side.ToString() + " - " + this.character.title + this.character.name);
		data.Add("title", this.title);
		data.Add("damage", this.character._hurt.ToString());
		
		BG.addMessage("damage", data);
	}
}

File: website/lib/class/debuff/debuff_2.php
<?php
class debuff_2 extends debuff_prototype{
	public $id = 2;
	public $title = '燃燒';
	public $type = 'end';
	public $duration = 3;
	public $orgDuration = 3;
	public $isStacked = true;
	public $stackMaxCount = 3;

	function effect(){
		$damage = round($this->caster->mp * 0.7) * $this->stackCount;
		$damage = $this->caster->calcDamage($damage, $this->char, 2);
		$this->char->dot($damage, 2);	//dot傷害第二個值要帶true

		$data = array(
			'prop'			=>	2,
			'targetId'		=>	$this->char->side . '-' . $this->char->charId,
			'targetTitle'	=>	$this->char->side . ' - ' . $this->char->title. $this->char->name,
			'title'			=>	$this->title,
			'damage'		=>	$this->char->hurt
		);
		battleground::addMessage('damage', $data);
	}
}

?>